google/cloud/security/actions/action_config_validator.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Actions configuration validator."""
+
+from collections import defaultdict
+import importlib
+import os
+import yaml
+
+class ActionsConfigValidator(object):
+    """ActionsConfigValidator class.
+
+    Contains basic validation that should be done on all action configurations.
+    """
+
+    @staticmethod
+    def validate(actions_config_path):
+        """Validates the actions config.
+
+        Action configuration validation:
+        1. Config is a valid yaml file.
+        2. Each action has an id and it is unique.
+        3. Check each action for:
+          1. An action type.
+          2. That the action type is valid.
+          3. A trigger.
+          4. That the trigger is valid.
+
+        Args:
+            actions_config_path (str): The actions configuration path.
+
+        Returns:
+            tuple: A tuple with the dict configuration and the list of errors.
+
+        Raises:
+            ConfigLoadError: If the config cannot be loaded.
+        """
+        errors = []
+        config = ActionsConfigValidator._load_and_validate_yaml(
+            actions_config_path)
+
+        actions, errs = ActionsConfigValidator._load_actions(config)
+        errors.extend(errs)
+
+        for action in actions:
+            action_errs = []
+            action_errs.append(
+                ActionsConfigValidator._check_action_type(action))
+            action_errs.append(ActionsConfigValidator._check_trigger(action))
+            for err in action_errs:
+                if err is not None:
+                    errors.append(err)
+        return config, errors
```

Comment:
Does it make sense to use a go idiom here? I would except an exception with the list of errors as an attribute.